frontend/dockerfile/parser/directives_test.go

```diff
@@ -0,0 +1,105 @@
+package parser
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDirectives(t *testing.T) {
+	t.Parallel()
+
+	dt := `#escape=\
+# key = FOO bar
+
+# smth
+`
+
+	parser := DirectiveParser{}
+	d, err := parser.ParseAll([]byte(dt))
+	require.NoError(t, err)
+	require.Len(t, d, 1)
+
+	require.Equal(t, d[0].Name, "escape")
+	require.Equal(t, d[0].Value, "\\")
+
+	// for some reason Moby implementation in case insensitive for escape
+	dt = `# EScape=\
+# KEY = FOO bar
```

Comment:
Yeah, this was a clear case of "fatigue" kicking in after a _very_ long debate about the feature (it originally was implemented as a new `ESCAPE` Dockerfile instruction). Once we got to the concept of "parser directives", there was debate about "upper" or "lowercase"; https://github.com/moby/moby/pull/22268#issuecomment-220152257, and in the end it was decided to "recommend" lowercase, but "allow" uppercase https://github.com/moby/moby/pull/22268#issuecomment-220349091 https://github.com/moby/moby/pull/22268#issuecomment-220367638

I'd be open to (perhaps for Dockerfile instructions as well) discuss making it slightly more strict, and accept `lower` and `UPPER`, but not `mIxEDcAsE` (which we currently support in most cases), although we should look if there's actual uses of mixed-case and if it would be too much of a breaking change (we could "warn" ahead perhaps).